plasmapy/physics/distribution.py

```diff
@@ -370,7 +508,129 @@ def Maxwellian_speed_1D(v,
+                                        particle=particle,
+                                        method="most_probable")).si.value
+    # Get thermal velocity squared
+    vThSq = vTh ** 2
+    # Get square of relative particle velocity
+    vSq = (v - V_drift) ** 2
+    # calculating distribution function
+    coeff = 2 * (vThSq * np.pi) ** (-1 / 2)
+    expTerm = np.exp(-vSq / vThSq)
+    distFunc = coeff * expTerm
+    if units == "units":
+        return distFunc.to(u.s / u.m)
+    elif units == "unitless":
+        return distFunc
+
+
+def Maxwellian_speed_2D(v,
+                        T,
+                        particle="e",
+                        V_drift=0,
+                        vTh=np.nan,
+                        units="units"):
+    r"""
+    pdf of speed for a 2D Maxwellian distribution.
+
+    Return the probability of finding a particle with speed components
+    `vx` and `vy` in m/s in an equilibrium plasma of temperature
+    `T` which follows the 2D Maxwellian distribution function. This
+    function assumes Cartesian coordinates.
+
+    Parameters
+    ----------
+    v: ~astropy.units.Quantity
+        The speed in units convertible to m/s.
+
+    T: ~astropy.units.Quantity
+        The temperature, preferably in Kelvin.
+
+    particle: str, optional
+        Representation of the particle species(e.g., `'p'` for protons, `'D+'`
+        for deuterium, or `'He-4 +1'` for :math:`He_4^{+1}`
+        (singly ionized helium-4)), which defaults to electrons.
+
+    V_drift: ~astropy.units.Quantity
+        The drift speed in units convertible to m/s.
+
+    vTh: ~astropy.units.Quantity, optional
+        Thermal velocity (most probable) in m/s. This is used for
+        optimization purposes to avoid re-calculating vTh, for example
+        when integrating over velocity-space.
+
+    units: str, optional
+        Selects whether to run function with units and unit checks (when
+        equal to "units") or to run as unitless (when equal to "unitless").
+        The unitless version is substantially faster for intensive
+        computations.
+
+    Returns
+    -------
+    f : ~astropy.units.Quantity
+        Probability in speed^-1, normalized so that:
+        :math:`\iiint_{0}^{\infty} f(\vec{v}) d\vec{v} = 1`.
+
+    Raises
+    ------
+    TypeError
+        A parameter argument is not a `~astropy.units.Quantity` and
+        cannot be converted into a `~astropy.units.Quantity`.
+
+    ~astropy.units.UnitConversionError
+        If the parameters is not in appropriate units.
+
+    ValueError
+        If the temperature is negative, or the particle mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+    In 2D, the Maxwellian speed distribution function describing
+    the distribution of particles with speed :math:`v` in a plasma with
+    temperature :math:`T` is given by:
+
+    .. math::
+
+       f = 2 \pi \vec{v} (\pi v_{Th}^2)^{-1} \exp(-(\vec{v} -
+       \vec{V}_{drift})^2 / v_{Th}^2)
```

Comment:
I don't think this is correct; when the drift speed is not zero, it is non-trivial to integrate over angles to get a probability density function for speed; see my working here: https://www.dropbox.com/s/rsjb65nvl2v9kbh/max_2d.pdf?dl=0

Reply:
Doesn't the drift velocity just shift the position of distribution in velocity space? (thinking in Cartesian coordinates). Just thinking about it spatially/geometrically, this shouldn't affect the normalization coefficient.

Thinking about it numerically, if you want to get the normalization coefficient you can just make the substitution `v_new = v_old - v_drift`. Since `v_drift` is a constant we have `dv_new = dv_old` and we can just do the integration as we would in the case of `v_drift = 0`.

The one caveat here is that I think it modifies the velocity terms in the normalization coefficient. So for a 3D distribution we have `4 * pi * v ** 2` representing the spherical shell over which we are integrating. I think this shell gets shifted to be centered around the drifting Maxwellian, so it would be something like  `4 * pi * (v - v_drift) ** 2`.

Reply:
It doesn't affect the normalisation coefficient, but it does effect the probability distribution function (pdf) of the radial velocity coordinate (ie speed) in a polar coordinate system - see my above derivation. Let me know if anything doesn't make sense, but I'm pretty sure that with a drift velocity the pdf of speed currently in the code is wrong.

Reply:
Yea I've taken a look at the derivation and I'm not convinced. What you argue in the derivation would imply an effect on the normalization coefficient (since the exponentials cannot be affected because they're exponentials), otherwise nothing is affected and the current implementation is correct.

I'll try to work this out on paper using change of variables.

Reply:
My point is that doing the integral in my equation (6) isn't possible (or at least doesn't come out with the equation currently in the docstring) for v_drift != 0.

If v_drift = 0 the integral is possible, and results in the equation in the docstring (with v_drift = 0).

Reply:
I've added a bit at the end with v_drift = 0, to show how it reduces to the expected equation which might help?

Reply:
So here is my derivation for the current implementation of the velocity and speed distribution including drift. This is using a change of variables to shift the velocity distribution to the origin, do the integration, and then shift back the normalized distribution(s).

![img_20180511_024812](https://user-images.githubusercontent.com/28945309/39915837-38aed98a-54c6-11e8-9f61-59c1d18cdbdf.jpg)

Reply:
I don't think the final equation can be correct, as it goes negative for v > v_drift (and probability densities can't be negative!). I think the problem is that the final "shifting back" stage is not valid - shifting in velocity space is not the same as shifting in speed (radial coordinate) space.

Reply:
Yeah, in that last step, you have to take into account the differential bit when changing coordinates, ie. 
![image](https://user-images.githubusercontent.com/6197628/39916752-5a7625e0-5503-11e8-9539-50fa60e4241b.png)
 and I think you have assumed that dv/dv' = 1, which isn't true.

Reply:
hmmm, yea it goes negative for `v < v_drift`... perhaps the normalization constant is just `v` then, since the size of the area of integration doesn't change due to drift offsetting the distribution in velocity space.

Good point about shifting in velocity vs in speed space. There must be a way to convert the shift in velocity space to one in speed space. I think that will involve an absolute value since `v = |\vec{v}|`... so maybe the `v - v_drift` part of the normalization coefficient is wrapped in an absolute value. 

I'm also digging into your approach and seeing if I can crack it with some trig identities. If you don't assume drift velocity along one axis, and instead keep it general, you end up with a fairly compact form for the velocity terms:
`v_r ** 2 - 2 * v_r * v_drift_r * cos(v_phi - v_drift_phi) + v_drift_r **2`

You could then integrate each of the 3 terms separately.

Reply:
I would like to note, it's astounding how I can't find a single page on the internet that discusses this problem :sob:

Reply:
Yep, I agree that in theory you might be able to integrate those three terms (you only actually need to integrate the middle one, since that's the only one that depends on v_phi. It's possible, and you get a modified Bessel function: https://www.wolframalpha.com/input/?i=integrate+e%5E(-a*cos(x))+between+0+and+2pi

Reply:
So my conclusion is that the function and docstring equation should have a modified Bessel function in it somewhere.

Reply:
So I'm working this out and am having trouble getting the normalization constant in the case of non-zero drift. That damn Bessel function has a radial velocity and so it shows up in the radial integral from 0 to Inf. [Wolfram doesn't know what to do with it](https://www.wolframalpha.com/input/?i=integrate+x+*+e%5E(-x%5E2+%2F+a%5E2)+*+I_0(b+*+x)+between+0+and+inf)

Reply:
It looks like the 0th modified Bessel function of the first kind has a [nice series expansion](https://reference.wolfram.com/language/ref/BesselI.html), this would then yield a series of integrals which are just different moments of `v`. Then perhaps there is a way to return the function to some sensible form, maybe another Bessel function? Just some ponderings...

Reply:
Ooh, I found the first mention of Bessel functions and probability distribution functions! Apparently the p.d.f. [follows a Ricean probability density function](http://elec424.wikia.com/wiki/Modified_Bessel_Functions). Funny how once you find the right jargon you start finding all sorts of [links](https://en.wikipedia.org/wiki/Rice_distribution).

Reply:
Might be able to use [SciPy's Rice p.d.f. function](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rice.html).